lisa/features/non_ssh_executor.py

```diff
@@ -0,0 +1,50 @@
+from typing import List
+
+from lisa.feature import Feature
+from lisa.features.serial_console import SerialConsole
+
+
+class NonSshExecutor(Feature):
+    """
+    NonSshExecutor is used to run commands on the node when SSH is not available.
+    Lisa by default uses SSH for connection, but this feature provides an alternative
+    execution method for scenarios where SSH connectivity is not possible or desired.
+    """
+
+    @classmethod
+    def name(cls) -> str:
+        return "NonSshExecutor"
+
+    def enabled(self) -> bool:
+        return True
+
+    def execute(self, commands: List[str]) -> List[str]:
+        """
+        Executes a list of commands on the node and returns their outputs.
+
+        :param commands: A list of shell commands to execute.
+        :return: A string containing the output of the executed commands.
+        """
+        out = []
+        if not self._node.features.is_supported(SerialConsole):
+            raise NotImplementedError(
+                "NonSshExecutor requires SerialConsole feature to be supported."
+            )
+        serial_console = self._node.features[SerialConsole]
+        serial_console.ensure_login()
+        # clear the console before executing commands
+        serial_console.write("\n")
+        _ = serial_console.read()
+        for command in commands:
+            serial_console.write(self._add_newline(command))
+            out.append(serial_console.read())
```

Comment:
besides return the output, also printed them out.

Reply:
def _collect_logs_using_non_ssh_executor (the caller) is currently printing it.

Should I add an optional argument to print it in this function itself?

Reply:
The method should print by itself, instead of caller. If it's used by others, they don't need to copy code to print. It's similar to how node.execute print out.